### PR TITLE
[AKS] Add node image only option for CLI

### PR DESCRIPTION
**Pull request by gtxistxgao**
**Description**  

To GA node image upgrade, we need to add support on CLI. This PR adds the ability to use following commands to upgrade node image:

- Upgrade the node image version of all node pools in a cluster : `az aks upgrade -g resourceGroupName -n clusterName --node-image-only`

- Upgrade one node pool's node image version: `az aks nodepool upgrade -g resourceGroupName --cluster-name clusterName -n nodePoolName --node-image-only`

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2904,12 +2934,25 @@ def aks_agentpool_scale(cmd, client, resource_group_name, cluster_name,
 
 
 def aks_agentpool_upgrade(cmd, client, resource_group_name, cluster_name,
-                          kubernetes_version,
                           nodepool_name,
+                          kubernetes_version='',
+                          node_image_only=False,
                           no_wait=False):
     instance = client.get(resource_group_name, cluster_name, nodepool_name)
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2941]: These 2 lines can be moved below the `node_image_only` part.

**Review by arrownj — COMMENTED** [review]
Hi @gtxistxgao , just as other PRs, as this is a new option, could you please also add tests for it ? I think there would be an existing upgrade test case already, maybe we just add add this option in this case.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (gtxistxgao) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: make sense. moved!

**Comment by gtxistxgao** [comment]
> Hi @gtxistxgao , just as other PRs, as this is a new option, could you please also add tests for it ? I think there would be an existing upgrade test case already, maybe we just add add this option in this case.

Hi, test added. There is no existing one so I create 2 more for this.

**Review by arrownj — APPROVED** [review]

**Pull request merged**